src/components/SystemPages/NewReaderControls.vue

```diff
@@ -0,0 +1,377 @@
+<template>
+  <div class="newReaderControls">
+    <template v-if="featureList.includes('pagenumber')">
+      <!-- <span v-if="$localData.settings['devMode']" v-text="newReaderPage" /> -->
+      <div class="newReaderInput">
+        <!-- Settings for adjusting new reader mode -->
+        <template v-if="$isNewReader">
+          <p>New reader mode enabled.</p>
+          <!-- Can't show picker if you're in viz mode. -->
+          <StoryPageLink
+            v-if="isNewReadingPreHS" 
+            :mspaId='$newReaderCurrent'/>
+          <input type="number" size="1" maxlength="6" 
+            v-model="newReaderPageInput"
+            v-else
+            @keydown.enter="changeNewReader()"
+            :class="{
+              invalid: !isValidPageSet, 
+              empty: !newReaderPageInput || !newReaderPageInput.length, 
+              changed: newReaderPageChanged
+            }" >
+            <span :class="{ urgent: newReaderPageChanged }">
+            <button :disabled="!(isValidPageSet && newReaderPageChanged)" 
+              @click="changeNewReader()" 
+              class="rightBtn"
+              >Adjust</button>
+            </span>
+        </template>
+        <template v-else>
+          <p>New reader mode disabled.</p>
+          <input type="number" size="1" maxlength="6" 
+            v-model="newReaderPageInput"
+            @keydown.enter="setNewReader()"
+            :class="{
+              invalid: !isValidPageSet, 
+              empty: !newReaderPageInput || !newReaderPageInput.length, 
+              changed: newReaderPageChanged
+            }" >
+          <span :class="{ urgent: newReaderPageChanged }">
+          <button :disabled="!isValidPageSet || newReaderPageInput.length < 1" 
+          @click="setNewReader()"
+          class="rightBtn">Activate</button>
+          </span>
+        </template>
+        <p class="hint" v-if="$localData.settings.mspaMode">
+          Enter an <strong>MS Paint Adventures</strong> page number<br>
+          e.g. www.mspaintadventures.com/?s=6&p=<strong>004130</strong></p>
+        <p class="hint" v-else>
+          Enter a <strong>Homestuck.com</strong> page number between 1 and 8129.<br>
+          e.g. www.homestuck.com/story/<strong>413</strong></p>
+        <div style="height: 40px;" v-if="$isNewReader">
+            <br />
+          <div class="bigButtonRow">
+            <button @click="clearNewReader()">Switch off new reader mode</button>
+          </div>
+        </div>
+        <div style="height: 40px;" v-else>
+          <p class="hint">Or activate a preset:</p>
+          <div class="bigButtonRow">
+            <button @click="setupProblemSleuth()">Start Problem Sleuth</button>
```

Comment:
Switching from MSPA numbers to Viz numbers while a non-homestuck page is selected freaks the component out.
![image](https://user-images.githubusercontent.com/28245739/152683140-725837c5-5f99-44e5-8272-fa5fc33d24eb.png)

Reply:
Fixed